### PR TITLE
Remove analytics calls for temporary basket calculation

### PR DESCRIPTION
**Pull request by BenjiLee**
https://openedx.atlassian.net/browse/LEARNER-4968

### CODE REVIEW

File: ecommerce/extensions/basket/models.py [hunk 1]
```diff
@@ -49,13 +51,14 @@ def get_basket(cls, user, site):
...
+        cached_response = RequestCache.get_cached_response(TEMPORARY_BASKET_CACHE_KEY)
+        if cached_response.is_miss:
+            for line in self.all_lines():
+                # Do not fire events for free items. The volume we see for edX.org leads to a dramatic increase in CPU
+                # usage. Given that orders for free items are ignored, there is no need for these events.
+                if line.stockrecord.price_excl_tax > 0:
+                    properties = translate_basket_line_for_segment(line)
+                    track_segment_event(self.site, self.owner, 'Product Removed', properties)
 
         super(Basket, self).flush()  # pylint: disable=bad-super-call
```
Comment (BenjiLee) [inline ecommerce/extensions/basket/models.py]: I'd love to know why this was pylint ignored and at the bottom :/

File: ecommerce/extensions/basket/constants.py [hunk 2]
```diff
@@ -0,0 +1 @@
+IS_TEMPORARY_BASKET_CACHE_KEY = "Calculate temporary basket"
```
Comment (robrap) [inline ecommerce/extensions/basket/constants.py]: Nit: Can we name this something more like: `ecommerce.is_calculate_temporary_basket`?

1. Namespace is not important for request cache, but it’s a good habit.
2. I’d rather the key look more like a key than a description.

File: ecommerce/extensions/basket/models.py (hunk 1 above)
Comment (robrap) [reply ecommerce/extensions/basket/models.py]: On your pylint note, are there any clues? Who wrote it? (Reviewing by phone.)

**Review by robrap — APPROVED** [review]
1. Can you add a note to JIRA to clean this up if we can stop using the basket for calculate?
2. Can you add an assertion that track_segment_event is never called? It should fail without your changes (temporary) and pas with your changes.

Approval conditional on minor requests.

File: ecommerce/extensions/basket/models.py (hunk 1 above)
Comment (BenjiLee) [reply ecommerce/extensions/basket/models.py]: https://github.com/edx/ecommerce/pull/1386 Everyone is gone >.< I think its safe to move it back up. It looks like she someone commented to put the super method back in, but never specified where, only to put it in.

Comment (robrap) [reply ecommerce/extensions/basket/models.py]: Ok.

Comment (BenjiLee) [reply ecommerce/extensions/basket/models.py]: nvm. we need this down there

**Comment by BenjiLee** [comment]
This is the note for cleanup https://openedx.atlassian.net/browse/LEARNER-5463

File: ecommerce/extensions/basket/models.py (hunk 1 above)
Comment (robrap) [reply ecommerce/extensions/basket/models.py]: If you understand now, and you didn't before, does it warrant a comment to explain?

File: ecommerce/extensions/basket/constants.py [hunk 3]
```diff
@@ -0,0 +1 @@
+is_calculate_temporary_basket = "Calculate temporary basket"
```
Comment (robrap) [inline ecommerce/extensions/basket/constants.py:1]: I'm sad this didn't change. :(

Comment (BenjiLee) [reply ecommerce/extensions/basket/constants.py:1]: Ah! I thought I made the change you wanted. I must have misunderstood

Comment (BenjiLee) [reply ecommerce/extensions/basket/constants.py:1]: > 2. I’d rather the key look more like a key than a description.

Looks like I didn't do this part.

> Nit: Can we name this something more like: ecommerce.is_calculate_temporary_basket?
 
and did this instead.

Comment (robrap) [reply ecommerce/extensions/basket/constants.py:1]: I meant:
```
is_calculate_temporary_basket = "ecommerce.is_calculate_temporary_basket"
```

**Pull request merged**